### PR TITLE
Add draft decision about Braze Event Delivery

### PR DESCRIPTION
**Pull request by cpennington**
**Description:** ADR for Braze Event Delivery

**JIRA:** https://openedx.atlassian.net/browse/AA-667

**Dependencies:** N/A

**Merge deadline:** N/A

**Installation instructions:** List any non-trivial installation 
instructions.

**Testing instructions:**

1. Open page A
2. Do thing B
3. Expect C to happen
4. If D happened instead - check failed.

**Reviewers:**
- [ ] tag reviewer 
- [ ] tag reviewer 

**Merge checklist:**
- [ ] All reviewers approved
- [ ] CI build is green
- [ ] Version bumped
- [ ] Changelog record added
- [ ] Documentation updated (not only docstrings)
- [ ] Commits are squashed

**Post merge:**
- [ ] Create a tag
- [ ] Check new version is pushed to PyPI after tag-triggered build is 
      finished.
- [ ] Delete working branch (if not needed anymore)

**Author concerns:** List any concerns about this PR - inelegant 
solutions, hacks, quick-and-dirty implementations, concerns about 
migrations, etc.

### CODE REVIEW

File: docs/decisions/0001-braze-event-delivery.rst [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+Status
+------
+
+Draft
+
+Context
+-------
+
+With the existing Sailthru ACE channel, edX's marketing team has limited visibility into
+the messages being sent out via ACE, both the quantity and the performance of those messages.
```
Comment (mikix) [inline docs/decisions/0001-braze-event-delivery.rst]: This is actually not true anymore with Braze - they know click rates and open rates and all that for each type of nudge email we send out.

That said, the rest of this ADR is about letting them change content, so maybe just switch this sentence to "With the existing ACE channels ... can't control content easily" or something.

File: docs/decisions/0001-braze-event-delivery.rst [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+In order to improve their ability to improve edX's marketing content, we want to expose the
+contents and performance of the marketing flows that are powered by ACE to that team.
+
+When exposing those data to the marketing organization, and allowing them to make changes
+to improve the content, we would like to make sure that they aren't blocked by the engineering
+organization. However, we also need to be mindful of existing open-source uses of ACE as
+well, and that we don't lose the ability to maintain those uses.
+
+While looking at Braze functionality, it was determined that there isn't a good way for us
```
Comment (mikix) [inline docs/decisions/0001-braze-event-delivery.rst:22]: There's a lot of Braze and edX marketing talk, in this open edx ADR. Is there a way to make this more generic? Maybe it is suitable for a `Context` section, I guess? But in the `Decision`, you are still talking about `We` == edx.org and marketing and such. Is this doc better served as an internal wiki page?

**Review by mikix — APPROVED** [review]
Seems good

File: docs/decisions/0001-braze-event-delivery.rst (hunk 1 above)
Comment (cpennington) [reply docs/decisions/0001-braze-event-delivery.rst]: Is each type of ACE email broken out into a separate Braze campaign that we can measure the performance of? That's super helpful even without the event-level mapping.

File: docs/decisions/0001-braze-event-delivery.rst (hunk 2 above)
Comment (cpennington) [reply docs/decisions/0001-braze-event-delivery.rst:22]: Well, in this case, the Braze channel is in this open-edx repo, so it seems like the ADR should go there as well. I'm fine w/ documenting the decision as being based on how edx.org is using that channel. Alternately, we could put the braze channel in a separate repo.

File: docs/decisions/0001-braze-event-delivery.rst (hunk 1 above)
Comment (mikix) [reply docs/decisions/0001-braze-event-delivery.rst]: Yeah - see https://github.com/edx/edx-internal/blob/master/edx-remote-config/prod/lms.yml#L13

If you look in Braze for a "ACE Dynamic Pacing" campaign, you can see the performance.

File: docs/decisions/0001-braze-event-delivery.rst (hunk 2 above)
Comment (mikix) [reply docs/decisions/0001-braze-event-delivery.rst:22]: Well an open edx installation might also be using Braze. It's appropriate code to live in an open repo. I was just suggesting we generic-ize this ADR or something. But ultimately, it doesn't really matter.

File: docs/decisions/0001-braze-event-delivery.rst (hunk 1 above)
Comment (cpennington) [reply docs/decisions/0001-braze-event-delivery.rst]: Oh, nice, I didn't realize we were already splitting out the templates.

Comment (cpennington) [reply docs/decisions/0001-braze-event-delivery.rst]: I see. They're being marked as separate variations in the same campaign...

I don't know, conceptually, if that's a coherent way of doing it (in Braze, variations are more for A/B testing, I think), but as long as marketing is getting what they need out of it, that's fine. I assume that means every new email added to ACE will also need a corresponding change to Braze and configuration?

Comment (cpennington) [reply docs/decisions/0001-braze-event-delivery.rst]: (That's a departure from how Sailthru works, where you can just add a new email and it'll send, but maybe that's an acceptable price to pay)

Comment (mikix) [reply docs/decisions/0001-braze-event-delivery.rst]: > I don't know, conceptually, if that's a coherent way of doing it (in Braze, variations are more for A/B testing, I think)

Ah that was just how I did it. It's a simple config change to make them different campaigns instead. Whatever marketing wants there.

> I assume that means every new email added to ACE will also need a corresponding change to Braze and configuration? 
> (That's a departure from how Sailthru works, where you can just add a new email and it'll send, but maybe that's an acceptable price to pay)

You can still just add a new email template to ACE and send. We don't require a configured campaign. But if you define one, we track it.

Comment (cpennington) [reply docs/decisions/0001-braze-event-delivery.rst]: Ah, nice work!

File: docs/decisions/0001-braze-event-delivery.rst (hunk 2 above)
Comment (cpennington) [reply docs/decisions/0001-braze-event-delivery.rst:22]: Ok, I tried to more explicitly name the relevant parties (edX's engineering and marketing teams).

File: docs/decisions/0001-braze-event-delivery.rst (hunk 1 above)
Comment (mikix) [reply docs/decisions/0001-braze-event-delivery.rst]: Christa also preferred the ACE emails as separate campaigns - they are now. Just search for ACE in braze and there should be 5 now

**Review by mikix — APPROVED** [review]
Odd that project coverage went down. We include rst files? Seems wrong.

**Pull request merged**